### PR TITLE
Fixed a few deprecations

### PR DESCRIPTION
**Pull request by TacoTheDank**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradlew app:connectedGmsDebugAndroidTest` to make sure you didn't break anything (with an emulator running or phone connected). I ran `./gradlew app:assembleFdroidDebugAndroidTest`, and it was successful. I can't run any emulator tests, so that's the best I've got.

- [x] If you have multiple commits please combine them into one commit by squashing them. I don't know how to do it with git, but GitHub lets you do it when merging.

- Convert preference to AndroidX preference (fixes deprecations, as regular preference was deprecated in API 29)
- Fix a couple miscellaneous other deprecations
- Use ContextCompat.getSystemService()

### CODE REVIEW

File: app/src/main/java/com/google/android/stardroid/touch/MapMover.java [hunk 1]
```diff
@@ -43,7 +44,9 @@ public MapMover(AstronomerModel model, ControllerGroup controllerGroup, Context
     this.controllerGroup = controllerGroup;
     Display display = ((WindowManager) context.getSystemService(
         Context.WINDOW_SERVICE)).getDefaultDisplay();
-    int screenLongSize = display.getHeight();
+    Point point = new Point();
+    display.getSize(point);
+    int screenLongSize = point.y;
```
Comment (jaydeetay) [inline app/src/main/java/com/google/android/stardroid/touch/MapMover.java]: Hi! What are you doing here? The rest of the PR looks good but I'm puzzled by this bit.

[Sorry it took a long time to respond - I wrote this comment ages ago and then didn't hit "submit"]

Comment (TacoTheDank) [reply app/src/main/java/com/google/android/stardroid/touch/MapMover.java]: Sorry, I don't know what this was about. This was wrong. I've fixed it with a push.

**Comment by TacoTheDank** [comment]
@jaydeetay 

So I've made some changes. They're all good as far as I can tell.

But there's a crash happening when I launch a built APK (that I don't believe has to do with the changes I made, as the files the crashes are located in aren't touched or related to the ones I did, I don't think):

```
FATAL EXCEPTION: pool-2-thread-1
Process: com.google.android.stardroid, PID: 13753
android.content.res.Resources$NotFoundException: String resource ID #0x7f0c01d4
	at android.content.res.Resources.getText(Resources.java:402)
	at android.content.res.Resources.getString(Resources.java:495)
	at com.google.android.stardroid.source.proto.ProtobufAstronomicalSource.getLabels(ProtobufAstronomicalSource.java:111)
	at com.google.android.stardroid.layers.AbstractSourceLayer.initialize(AbstractSourceLayer.java:74)
	at com.google.android.stardroid.layers.AbstractFileBasedLayer.access$201(AbstractFileBasedLayer.java:46)
	at com.google.android.stardroid.layers.AbstractFileBasedLayer$1.run(AbstractFileBasedLayer.java:65)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
```
I have no idea why this is happening.

**Comment by jaydeetay** [comment]
Don't worry too much about the crash. It's a consequence of our weird build system (which I'm slowly overhauling). I know how to fix it if the crash repeats for me.
Thanks for the deprecation fixes.

**Comment by jaydeetay** [comment]
I fixed the crash caused by the out-of-date resources, but I'm seeing a crash when I open the settings page. Logcat doesn't really tell me much:

10-04 21:35:03.889 16834 16834 E AndroidRuntime: Process: com.google.android.stardroid, PID: 16834
10-04 21:35:03.889 16834 16834 E AndroidRuntime:        at com.google.android.stardroid.activities.EditSettingsActivity.onStart(EditSettingsActivity.java:86)

which is the call to the superclass's onStart method.

**Comment by jaydeetay** [comment]
I just reverted the EditSettingsActivity change for now as it's crashing onStart and I don't know why.

**Comment by TacoTheDank** [comment]
@jaydeetay I'll try to tackle it.

**Comment by jaydeetay** [comment]
Thanks.  The logs didn't really give much away. I only tried it on a Pixel
4, btw.

**Comment by TacoTheDank** [comment]
@jaydeetay I can't catch the settings crash because the launch crash from before is still occurring, and is preventing me from getting to it :/

You seem to be able to launch the app, so could you grab the crash using [Scoop](https://forum.xda-developers.com/android/apps-games/adb-root-scoop-catch-stack-trace-app-t3888798), please? (Scoop is awesome, I recommend its usage in general).

**Comment by jaydeetay** [comment]
I'll try Scoop.

To fix the crash-on-launch try using the build_skymap.sh script which will rewrite the datafiles in a compatible way with the binary. It could be that your build is generating different resource ids to mine for some reason.

**Comment by TacoTheDank** [comment]
@jaydeetay I get multiple build failures running the script. First is that `build/install/datagen/bin/datagen` doesn't exist. I tried changing Gms to Fdroid and that fixed the missing google-services.json thing, but several other things are failing, with vague stuff like "compileJava failed", the datagen still not existing, and stuff like that :/

**Pull request merged**